### DISCUSSION TITLE
Best way to update this entire child relation? DeleteMany then Create?

### DISCUSSION THREAD

**Discussion by justin-harr** [prisma/prisma#13891] [Q&A]
I'm using Prisma with Planetscale. I have two models, Palette and Swatch. Palettes can have many Swatches, Swatches can only belong to one Palette. Also, Swatches are only interacted with on the Palette level, you can't fetch/update/delete an individual Swatch, only update the whole Palette at once. Here are the relevant models:

```prisma
model Palette {
  id          String   @id @default(cuid())
  name        String
  swatches    Swatch[]
}

model Swatch {
  id        Int     @id @default(autoincrement())
  name      String
  color     String
  paletteId String
  palette   Palette @relation(fields: [paletteId], references: [id], onDelete: Cascade)

  @@index([paletteId])
}
```
Here is a screenshot to help visualize what I'm modeling:
<img width="1143" alt="Screen Shot 2022-06-18 at 4 56 52 PM" src="https://user-images.githubusercontent.com/28981975/174456861-6c5cbf7f-b425-47e0-8979-e35b7c9836ee.png">

When I create a Palette, all of the relevant data is sent and I create a Palette record like this with all Swatches included:

```ts
export const createPalette = async ({ name, swatches }) =>
  prisma.palette.create({
    data: {
      name,
      swatches: {
        create: swatches,
      },
    },
    include: {
      swatches: true,
    },
  })
```

This works great. Then when a user edits a Palette the same screen is rendered and a user can update the Palette name, as well as create new Swatches, update existing ones, delete existing ones, and rearrange Swatches. Then all of the data is sent at the same time to the backend again.

I'm trying to figure out the best way to updated the Palette's Swatches since they could be added, updated, deleted, or moved all at once. So far the best I have is to delete all of the swatches and then create new ones like this?

```ts
export const updatePalette = async ({ paletteId, name, swatches }) => {
  await prisma.swatch.deleteMany({
    where: {
      paletteId,
    },
  })
  return prisma.palette.update({
    where: {
      id: paletteId,
    },
    data: {
      swatches: {
        create: swatches,
      },
    },
  })
}
```

But this seems wrong. I'd think `set` would be a better method but it doesn't create new items. Is there a better way to handle this?

**Comment by austincrim** [comment]
Hey @dasveloper 👋,

I think your solution is the way I would accomplish this as well. I would potentially wrap the update and delete operations in a [transaction](https://www.prisma.io/docs/concepts/components/prisma-client/transactions), though.

Another option would be to store the swatch information in a `Json` field and overwrite it each time, but you would lose type-safety and normalization.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
